### PR TITLE
feat(scan): add mutelist and config file to scan

### PR DESCRIPTION
**Pull request by pedrooot**
### Description

Mutelist and config file should be used for the Scan, if no files are provided the default ones will be used.

### Checklist

- Are there new checks included in this PR? Yes / No
    - If so, do we need to update permissions for the provider? Please review this carefully.
- [x] Review if the code is being covered by tests.
- [x] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [x] Review if backport is needed.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/aws_provider.py [hunk 1]
```diff
@@ -298,6 +302,10 @@ def scan_unused_services(self):
     def audit_config(self):
         return self._audit_config
 
+    @audit_config.setter
+    def audit_config(self, audit_config_path):
+        self._audit_config = load_and_validate_config_file("aws", audit_config_path)
```
Comment (jfagoagas) [inline prowler/providers/aws/aws_provider.py]:
```suggestion
        self._audit_config = load_and_validate_config_file(self._type, audit_config_path)
```

File: prowler/providers/azure/azure_provider.py [hunk 2]
```diff
@@ -184,7 +187,11 @@ def locations(self):
     @property
     def audit_config(self):
         """Returns the audit configuration for the Azure provider."""
-        return self._audit_config
+        return self._
+
+    @audit_config.setter
+    def audit_config(self, audit_config_path):
+        self._audit_config = load_and_validate_config_file("azure", audit_config_path)
```
Comment (jfagoagas) [inline prowler/providers/azure/azure_provider.py]:
```suggestion
        self._audit_config = load_and_validate_config_file(self._type, audit_config_path)
```

File: prowler/providers/gcp/gcp_provider.py [hunk 3]
```diff
@@ -168,6 +171,10 @@ def excluded_project_ids(self):
     def audit_config(self):
         return self._audit_config
 
+    @audit_config.setter
+    def audit_config(self, audit_config_path):
+        self._audit_config = load_and_validate_config_file("gcp", audit_config_path)
```
Comment (jfagoagas) [inline prowler/providers/gcp/gcp_provider.py]:
```suggestion
        self._audit_config = load_and_validate_config_file(self._type, audit_config_path)
```

File: prowler/providers/kubernetes/kubernetes_provider.py [hunk 4]
```diff
@@ -100,6 +103,12 @@ def namespaces(self):
     def audit_config(self):
         return self._audit_config
 
+    @audit_config.setter
+    def audit_config(self, audit_config_path):
+        self._audit_config = load_and_validate_config_file(
+            "kubernetes", audit_config_path
```
Comment (jfagoagas) [inline prowler/providers/kubernetes/kubernetes_provider.py]:
```suggestion
            self._type, audit_config_path
```

File: prowler/lib/scan/scan.py [hunk 5]
```diff
@@ -18,14 +19,23 @@ class Scan:
     _service_checks_completed: dict[str, set[str]]
     _progress: float = 0.0
     _findings: list = []
+    _mutelist_file: str = None
+    _config_file: str = None
```
Comment (jfagoagas) [inline prowler/lib/scan/scan.py]: Do we need to store them in the class? I mean because they are both stored in the provider.

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: prowler/lib/scan/scan.py (hunk 5 above)
Comment (pedrooot) [reply prowler/lib/scan/scan.py]: True! It's not needed

File: prowler/providers/kubernetes/kubernetes_provider.py (hunk 4 above)
Comment (pedrooot) [reply prowler/providers/kubernetes/kubernetes_provider.py]: Done

File: prowler/providers/gcp/gcp_provider.py (hunk 3 above)
Comment (pedrooot) [reply prowler/providers/gcp/gcp_provider.py]: Done

File: prowler/providers/azure/azure_provider.py (hunk 2 above)
Comment (pedrooot) [reply prowler/providers/azure/azure_provider.py]: Done

File: prowler/providers/aws/aws_provider.py (hunk 1 above)
Comment (pedrooot) [reply prowler/providers/aws/aws_provider.py]: Done

File: prowler/lib/scan/scan.py [hunk 6]
```diff
@@ -21,14 +21,15 @@ class Scan:
...
-
+      
     def __init__(self, provider: Provider, checks_to_execute: list[str] = None):
         """
         Scan is the class that executes the checks and yields the progress and the findings.
 
         Params:
             provider: Provider -> The provider to scan
             checks_to_execute: list[str] -> The checks to execute
+            mutelist_file: str -> The mutelist file (default: None)
```
Comment (jfagoagas) [inline prowler/lib/scan/scan.py]: Is this still needed?

File: prowler/providers/aws/aws_provider.py [hunk 7]
```diff
@@ -84,8 +90,9 @@ def __init__(
         scan_unused_services: bool = None,
         resource_tags: list[str] = [],
         resource_arn: list[str] = [],
-        audit_config: dict = {},
+        config_file: str = None,
```
Comment (jfagoagas) [inline prowler/providers/aws/aws_provider.py]: Why are you changing this? We should support passing the config as a `dict`.

File: prowler/providers/aws/aws_provider.py [hunk 8]
```diff
@@ -84,8 +90,9 @@ def __init__(
...
         fixer_config: dict = {},
+        mutelist_path: str = None,
```
Comment (jfagoagas) [inline prowler/providers/aws/aws_provider.py:100]: Thinking about this, maybe we can have `mutelist_path` and `mutelist_content` to allow the caller pass the config directly as an object.

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: prowler/providers/aws/aws_provider.py (hunk 8 above)
Comment (pedrooot) [reply prowler/providers/aws/aws_provider.py:95]: I don't get the point of this, inside `__init__` we implement the needed logic to parse the file to a dict so passing directly the dict won't add a value

File: prowler/lib/scan/scan.py (hunk 6 above)
Comment (pedrooot) [reply prowler/lib/scan/scan.py]: Good catch!

File: prowler/providers/aws/aws_provider.py (hunk 7 above)
Comment (pedrooot) [reply prowler/providers/aws/aws_provider.py]: True, good point

File: prowler/providers/aws/aws_provider.py (hunk 8 above)
Comment (pedrooot) [reply prowler/providers/aws/aws_provider.py:100]: Done

File: prowler/providers/aws/aws_provider.py [hunk 9]
```diff
@@ -281,10 +293,34 @@ def __init__(
         self._scan_unused_services = scan_unused_services
 
         # Audit Config
-        self._audit_config = audit_config
+        print(config_content)
+        print(config_path)
```
Comment (jfagoagas) [inline prowler/providers/aws/aws_provider.py]:
```suggestion
```

**Review by jfagoagas — DISMISSED** [review]
👏 thank you for this!

**Review by jfagoagas — DISMISSED** [review]

**Review by jfagoagas — APPROVED** [review]

**Pull request merged**